powerscale/helper/file_system_helper.go

```diff
@@ -281,7 +281,149 @@ func BuildFilesystemDatasource(ctx context.Context, state *models.FileSystemData
 }
 
 // UpdateFileSystemResourceState Updates File System Resource State.
-func UpdateFileSystemResourceState(ctx context.Context, plan *models.FileSystemResource, state *models.FileSystemResource, acl *powerscale.NamespaceAcl, meta *powerscale.NamespaceMetadataList) {
+func UpdateFileSystemResourceState(ctx context.Context, plan *models.FileSystemResource, acl *powerscale.NamespaceAcl, meta *powerscale.NamespaceMetadataList) (diags diag.Diagnostics) {
+
+	for _, attribute := range meta.Attrs {
+		switch *attribute.Name {
+		case "type":
+			plan.Type = types.StringValue(*attribute.Value)
+		case "create_time":
+			plan.CreationTime = types.StringValue(*attribute.Value)
+		}
+	}
+	if plan.Type.IsUnknown() {
+		plan.Type = types.StringNull()
+	}
+	if plan.CreationTime.IsUnknown() {
+		plan.CreationTime = types.StringNull()
+	}
+	if owner, ok := acl.GetOwnerOk(); ok {
+		ownerResType := types.StringNull()
+		if owner.Type != nil {
+			ownerResType = types.StringValue(owner.GetType())
+		}
+		ownerTypeUnknown := false
+		if plan.Owner.Type.IsUnknown() {
+			ownerTypeUnknown = true
+			plan.Owner.Type = ownerResType
+		} else if !ownerResType.Equal(plan.Owner.Type) {
+			diags.AddWarning("File System ACL Owner type is invalid", fmt.Sprintf("File System ACL Owner type is invalid, actual type is '%s'", ownerResType.ValueString()))
+		}
+		if !plan.Owner.ID.IsUnknown() && plan.Owner.Name.IsUnknown() {
```

Comment:
The owner and group part have several conditions and may easily get confused. Can we add some comment to help understand?

Also please rebase the `main` branch

Reply:
thanks @NavisJ , rebased and comments added; 
if still has confusion in comments please let me known, thanks